images/win/scripts/Installers/Install-MSYS2.ps1

```diff
@@ -0,0 +1,92 @@
+################################################################################
+##  File:  Install-MSYS2.ps1
+##  Desc:  Install MSYS2 and 64-bit gcc, cmake, & llvm (32-bit commented out)
+################################################################################
+
+# References
+# https://github.com/msys2/MINGW-packages/blob/master/azure-pipelines.yml
+# https://packages.msys2.org/group/
+
+$dash = "—"
+$wid = 80
+
+$orig_path = $env:PATH
+$git_path  = "C:\Program Files\Git"
+
+# get info from https://sourceforge.net/projects/msys2/files/Base/x86_64/
+$msy2_uri  = "http://repo.msys2.org/distrib/x86_64/msys2-base-x86_64-20190524.tar.xz"
+$msy2_sha1 = "cfe5035b1b81b43469d16bfc23be8006b9a44455".ToUpper()
+
+$msy2_file = "C:/msys2.tar.xz"
+
+# Download the latest msys2 x86_64
+Write-Host Starting download
+Invoke-WebRequest -UseBasicParsing -Uri $msy2_uri -OutFile $msy2_file
+Write-Host Finished download
+
+# check SHA1
+$dl_sha1 = (Get-FileHash -Path $msy2_file -Algorithm SHA1).Hash
+
+if ($dl_sha1 -ne $msy2_sha1) {
+  Write-Host "SHA1 mismatch, exiting"
+  exit 1
+}
+
+$msy2_file_u = "/$msy2_file".replace(':', '')
+
+# git has tar.exe in usr/bin, but xz.exe is in mingw64/bin
+$env:PATH = "$git_path\mingw64\bin;$orig_path"
+$tar      = "$git_path\usr\bin\tar.exe"
+
+# extract tar.xz to C:\
+&$tar -Jxf $msy2_file_u -C /c/
+
+Remove-Item $msy2_file
+
+Write-Host Finished extraction
+
+$env:PATH = "C:\msys64\mingw64\bin;C:\msys64\usr\bin;$orig_path"
```

Comment:
Am I right, that here you want to update PATH for the whole machine, and not for the current session?
If so, you can import the `ImageHelpers` module and use the `Add-MachinePathItem` method to update PATH.

Reply:
See below

Reply:
@AndreyMaslennikov 

> Am I right, that here you want to update PATH for the whole machine, and not for the current session?

Sorry, wasn't clear.  Just for the current session.